apps/cowswap-frontend/src/common/hooks/useAnnouncements.ts

```diff
@@ -0,0 +1,40 @@
+import { useAtomValue } from 'jotai'
+import { useMemo } from 'react'
+
+import { isProdLike } from '@cowprotocol/common-utils'
+import { Announcements, announcementsAtom } from '@cowprotocol/core'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+
+function useAnnouncements(chainId: SupportedChainId): Announcements {
+  const allAnnouncements = useAtomValue(announcementsAtom)
+
+  return useMemo(() => {
+    const env = isProdLike ? 'prod' : 'barn' // Should match what's set on CMS!
+
+    return allAnnouncements.reduce<Announcements>((acc, announcement) => {
+      if (
+        announcement.chainIds.length === 0 ||
```

Comment:
Please, split the condition into several consts with corresponding names

Reply:
1c992598f7ef